pepdbagent/utils.py

```diff
@@ -93,3 +94,16 @@ def tuple_converter(value: Union[tuple, list, str, None]) -> tuple:
     return tuple(
         " ",
     )
+
+
+def convert_date_string_to_date(date_string: str) -> datetime.datetime:
+    """
+    Convert string into datetime format
+
+    :param date_str: date string in format [YYYY:MM:DD]. e.g. 2022:02:22
+    :return: datetime format
+    """
+    try:
+        return datetime.datetime.strptime(date_string, "%Y:%m:%d") + datetime.timedelta(days=1)
+    except Exception:
+        raise IncorrectDateFormat
```

Comment:
I thought it's useful

Reply:
this seems like an error to me, since different types of exceptions could be caught here.

for example, if `datetime` is not imported or not found, then you'd be giving an `IncorrectDateFormat` error, which would give a red herring.

So in general, I think this is not good practice, as least not as implemented her.